logger.go

```diff
@@ -58,10 +62,18 @@ func Init(name string, verbose, systemLog bool, logFile io.Writer) {
 	}
 
 	flags := log.Ldate | log.Lmicroseconds | log.Lshortfile
-	infoLog = log.New(io.MultiWriter(iLogs...), "INFO: ", flags)
-	errorLog = log.New(io.MultiWriter(eLogs...), "ERROR: ", flags)
-	fatalLog = log.New(io.MultiWriter(eLogs...), "FATAL: ", flags)
-	initialized = true
+
+	var l logger
+	l.infoLog = log.New(io.MultiWriter(iLogs...), "INFO: ", flags)
+	l.errorLog = log.New(io.MultiWriter(eLogs...), "ERROR: ", flags)
+	l.fatalLog = log.New(io.MultiWriter(eLogs...), "FATAL: ", flags)
+
+	if !initialized {
```

Comment:
This global variable is now a little unfortunate, as it can race. (Which wasn't true before.) This needs to be guarded with a mutex. (Go is free to reorder the defaultLogger and initialized assignments as per the memory model as long as it doesn't change behavior within the same goroutine.)